Set cursor to start instead of empty range on clearHighlight

unhammer opened the pull request:
The highlight/selection seems to not get cleared by setting it to a new range. This change does `setCursor` instead with position at start of selection, which seems to clear the selection. 

(original version set it to start of doc, but that leads to scrolling if error is at e.g. page 2)

unhammer committed:
Set cursor to selection start instead of empty range on clearHighlight

dylanhand reviewed (approved):
If you can confirm it works, looks fine to me 👍

snomos commented:
I see that this PR is for the Google extension. Is there a similar problem on the MS Office side of things?

dylanhand commented:
@snomos let me know if you'd like me to re-deploy once this is merged.

snomos commented:
@dylanhand yes please do

unhammer commented:
It worked in my testing (though I kind of did it with manual google script pasting, I would feel more confident if I had tested a real "build" but I don't really know how to do that here).

I would guess the Microsoft API's are quite different? Unless they were made to mimic google's, which does not seem like a good idea.

unhammer commented:
In the currently deployed version, it's called, but it has no effect. Tested in both chrome and firefox.


https://github.com/user-attachments/assets/35b60355-e5c7-4329-bdbe-3e32bbca0669

snomos commented:
What did you expect should happen? That the selection should disappear as soon as the pointer is moved out of the error message box?

unhammer commented:
Yes. That's what the old code tries to do:

```js
                    $('.grammar-error').hover(highlightError, clearHighlight);
```
it's calling [jquery hover](https://api.jquery.com/hover/):

> .hover( handlerIn, handlerOut )
> Description: Bind two handlers to the matched elements, to be executed when the mouse pointer enters and leaves the elements.

So the `clearHighlight` function would be called, but didn't have the intended effect of clearing the highlight.

dylanhand committed:
Clear highlight sets cursor to end of highlighted word

IMO this feels more natural than putting the cursor at the beginning of the word.
